gems/smithy-xml/lib/smithy-xml/parser.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require_relative 'parser/frame'
+require_relative 'parser/stack'
+
+module Smithy
+  module Xml
+    # @api private
+    class Parser
+      # Raised when an XML parsing error occurs.
+      class ParseError < StandardError
+        def initialize(msg, line, column)
+          @line = line
+          @column = column
+          super(msg)
+        end
+
+        # @return [Integer, nil]
+        attr_reader :line
+
+        # @return [Integer, nil]
+        attr_reader :column
+      end
```

Comment:
We should probably move where we set the smithy-json's `ParserError` to keep these gems in a similar format if possible - https://github.com/smithy-lang/smithy-ruby/blob/decaf/gems/smithy-json/lib/smithy-json.rb#L13